### PR TITLE
Updated the check your answers page with constructor and methods.

### PR DESCRIPTION
**Pull request by DavidWoodward37**

### CODE REVIEW

File: test/constructors/nonresident/PersonalAndPreviousDetailsConstructorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,939 @@
...
+
+  "Calling .personalAndPreviousDetailsRows" should {
+
+    val result = PersonalAndPreviousDetailsConstructor.personalAndPreviousDetailsRows(Some(modelIndividual))
+
+    "when called with no model return an empty sequence" in {
+      PersonalAndPreviousDetailsConstructor.personalAndPreviousDetailsRows(None) shouldEqual Seq()
+    }
+
+    "when called with a valid model return a sequence of size 7" in {
```
Comment (TomDudley92) [inline test/constructors/nonresident/PersonalAndPreviousDetailsConstructorSpec.scala]: Should be size 8.

File: test/constructors/nonresident/AnswersConstructorSpec.scala [hunk 2]
```diff
@@ -293,7 +293,7 @@ class AnswersConstructorSpec extends UnitSpec with MockitoSugar {
     val constructor = setupMockedFinalAnswersConstructor(model)
 
     "when called with the model with all options return all options" in {
-      await(constructor.getPersonalDetailsAndPreviousCapitalGainsAnswers(hc)) shouldEqual model
+      await(constructor.getPersonalDetailsAndPreviousCapitalGainsAnswers(hc)) shouldEqual Some(model)
```
Comment (TomDudley92) [inline test/constructors/nonresident/AnswersConstructorSpec.scala]: Can you remove `import models.nonresident.{TotalPersonalDetailsCalculationModel, _}` and change it to `import models.nonresident._`

File: test/constructors/nonresident/PersonalAndPreviousDetailsConstructorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,939 @@
...
+
+      "not return some details for the PersonalAllowance" in {
+        result shouldEqual None
+      }
+    }
+  }
+
+  "calling .disabledTrusteeAnswerRow" when {
+
+    "a customer type of trustee and some income" should {
```
Comment (TomDudley92) [inline test/constructors/nonresident/PersonalAndPreviousDetailsConstructorSpec.scala]: please change to `and is claiming a disabled trustee`

File: test/constructors/nonresident/PersonalAndPreviousDetailsConstructorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,939 @@
...
+      s"return a link of ${routes.DisabledTrusteeController.disabledTrustee().url}" in {
+        result.fold(cancel("expected result not computed")) { item =>
+          item.link.fold(cancel("link not supplied when expected")) { link =>
+            link shouldBe routes.DisabledTrusteeController.disabledTrustee().url
+          }
+        }
+      }
+    }
+
+    "a customer type of anything but trustee and some income" should {
```
Comment (TomDudley92) [inline test/constructors/nonresident/PersonalAndPreviousDetailsConstructorSpec.scala]: please change to `and is claiming a disabled trustee`

File: test/controllers/CalculationControllerTests/CheckYourAnswersActionSpec.scala [hunk 5]
```diff
@@ -19,7 +19,7 @@ package controllers.CalculationControllerTests
 import constructors.nonresident.AnswersConstructor
 import controllers.helpers.FakeRequestHelper
 import controllers.nonresident.CheckYourAnswersController
-import models.nonresident._
+import models.nonresident.{TotalPersonalDetailsCalculationModel, _}
```
Comment (TomDudley92) [inline test/controllers/CalculationControllerTests/CheckYourAnswersActionSpec.scala]: Change to `import models.nonresident._`
Also remove unused imports.

File: app/controllers/nonresident/CheckYourAnswersController.scala [hunk 6]
```diff
@@ -66,13 +66,57 @@ trait CheckYourAnswersController extends FrontendController with ValidActiveSess
...
+  }
+
+  def checkAndGetFinalSectionsAnswers(totalGainResultsModel: TotalGainResultsModel,
+                                      calculationResultsWithPRRModel: Option[CalculationResultsWithPRRModel])(implicit hc: HeaderCarrier):
+  Future[Option[TotalPersonalDetailsCalculationModel]] = {
+
+    calculationResultsWithPRRModel match {
+
+      case Some(data) =>
+        val optionalSeq = Seq(calculationResultsWithPRRModel.get.rebasedResult, calculationResultsWithPRRModel.get.timeApportionedResult).flatten
```
Comment (DavidWoodward37) [inline app/controllers/nonresident/CheckYourAnswersController.scala]: change to data.rebasedResult and .timeApportionedResult

**Pull request merged**